### PR TITLE
Add History Multiview activity+panel, add "Show Recent" mode to multiview

### LINKED ISSUE #17143: Enhance MultiView by Displaying a Fixed Number of Recent Histories Instead of Always Tracking Pinned Histories, plus other improvements
As discussed [on Dec 07 2023 in galaxyproject/wg-ui-ux](https://matrix.to/#/!XciPHtOkuKNqKoVeKD:gitter.im/$7PiB7hfuavpasdi87FPhuBlKtzlrdlehbQ9CwQ5fP2Y?via=gitter.im&via=matrix.org), the Multi history view currently tracks histories you've loaded up previously. This can be annoying for users as this list of pinned histories is not updated for the histories created/updated recently, and a lot of clicks are needed to utilize this view.

Instead, it would be a more useful view if:
1. By default, the multi-history view is loaded up with the n most recently updated histories. 
    - The user can then, ofc, change the currently "selected" histories for this view, if they want
    - ~~In the case the user changes the currently selected histories, we start tracking the time this occurs, and whatever the currently selected histories are~~
    - ~~... until a certain time limit when we revert to the default n most recent histories.~~
    - In the case the user changes the currently selected histories, we start tracking them (like we already do).
    - Then in addition, we have a button somewhere in the MultiView for "Clear all from view" or "Show only recent", which is more explicit and though it requires a click, it is just one. (as recommended by @davelopez below)
2. Instead of using the modal to change the histories in view, or using the Right `HistoryPanel` to create histories when MultiView is active:
    - we move the History Selector (for the MultiView) to a fixed position in the left panel (?) (maybe in the form of an `ActivityBar` activity?)
    - and, add a `Create History` button in the left "selector" panel as well

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
Added a **History Multiview** activity and side panel, which can be used to select histories in Multiview.

This also adds the option of showing 4 latest histories in History multi view, instead of always tracking pinned histories. Once the user decides to track pinned histories, we do so, until the user "resets" it to recent histories mode.

Also organized the multiple histories in Multiview to show up evenly, with the `HistoryDetails` taking up a fixed space at the top.

Fixes https://github.com/galaxyproject/galaxy/issues/17143
Closes https://github.com/galaxyproject/galaxy/pull/16001 (every item when mounted, is loaded [here](https://github.com/galaxyproject/galaxy/pull/17251/files#diff-40d1262a87564bb81891c85dc43aed39e738dfbfeac61e4b0e7fee55f3a7527dR27))

https://github.com/galaxyproject/galaxy/assets/78516064/3d5c3609-7e96-4c22-bd7a-25c9bbeb50aa

## A singular `Histories` activity and panel?

This also opens up the potential of a singular `Histories` activity+side panel which - instead of changing route to the histories list (as the histories activity currently does) - toggles out a _Histories side panel_, and this panel contains:
1. A searchable History list (similar to the one in this Multiview panel) used to select the current history for the `HistoryPanel` (and other [AdditionalOptions](https://github.com/galaxyproject/galaxy/blob/a16c9c065a0fa0a78280b0cb8d0e615bedf1796f/client/src/components/History/Modals/SelectorModal.vue#L41)) **by default**
2. A button group used to either:
   - Open Histories list in center panel (side panel remains the same)
   - or, Open Multiview in center panel; then the History scroll list in side panel changes into the version from the Multiview panel, used to select histories for Multiview

Idea for https://github.com/galaxyproject/galaxy/issues/17284 ?

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

**Comment by ahmedhamidawan** [comment]
## A _Histories_ activity side panel
https://github.com/galaxyproject/galaxy/assets/78516064/b7155e95-08d4-49aa-bd3c-3345162eac08

This is separate from the `History Multiview` activity panel, and it can be used to quick operations on histories (like setting current, opening in center panel etc) and clicking on the `Histories` activity will not change user's route. To see all `Histories` in the center grid, the user can click on `All histories` in the footer (just like in the `NotificationsPanel`).

This can be tested here:
https://github.com/ahmedhamidawan/galaxy/tree/histories_side_panel

**Comment by ahmedhamidawan** [comment]
## A Singular _Histories_ activity and side panel
https://github.com/galaxyproject/galaxy/assets/78516064/6a6a9595-1c8b-43b6-b445-d3a0d10fc0a7

This is an example of a singular `Histories` activity and side panel (removes the `multiview` activity). It is a consolidation of the Histories and History Multiview side panel, and by default, it has the `Histories` side panel (from the example above). At the bottom, it has a button group which allows routing to either the Histories grid list or the multiview. When on the multiview route, the side panel switches to the multiview selector version, and on any other route it has the default, current history selector version.

This can be tested here:
https://github.com/ahmedhamidawan/galaxy/tree/singular_histories_side_panel

**Comment by ahmedhamidawan** [comment]
## Split Activity Buttons (panel and route both in one activity)
https://github.com/galaxyproject/galaxy/assets/78516064/ef7e4922-770b-4d41-89dc-b58c7299d487

This way the user can quickly open the History Grid or open the Histories panel

Implemented on top of https://github.com/ahmedhamidawan/galaxy/tree/histories_side_panel

File: client/src/components/Panels/MultiviewPanel.vue [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import FilterMenu from "@/components/Common/FilterMenu.vue";
+import HistoryList from "@/components/History/HistoryScrollList.vue";
+import ActivityPanel from "@/components/Panels/ActivityPanel.vue";
+
+const route = useRoute();
+const router = useRouter();
+
+// @ts-ignore bad library types
```
Comment (itisAliRH) [inline client/src/components/Panels/MultiviewPanel.vue]:
```suggestion
```

File: client/src/components/History/Multiple/MultipleViewList.vue [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
 import { useAnimationFrameScroll } from "@/composables/sensors/animationFrameScroll";
 import { Toast } from "@/composables/toast";
 import { useHistoryStore } from "@/stores/historyStore";
 import localize from "@/utils/localization";
+import { errorMessageAsString } from "@/utils/simple-error";
 
 import HistoryDropZone from "../CurrentHistory/HistoryDropZone.vue";
 import MultipleViewItem from "./MultipleViewItem.vue";
 
+// @ts-ignore bad library types
```
Comment (itisAliRH) [inline client/src/components/History/Multiple/MultipleViewList.vue]:
```suggestion
```

File: client/src/components/History/HistoryScrollList.vue [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+
+const props = defineProps({
+    multiple: { type: Boolean, default: false },
+    selectedHistories: { type: Array as PropType<PinnedHistory[]>, default: () => [] },
+    additionalOptions: { type: Array as PropType<AdditionalOptions[]>, default: () => [] },
+    showModal: { type: Boolean, default: false },
+    inModal: { type: Boolean, default: false },
+    filter: { type: String, default: "" },
+    loading: { type: Boolean, default: false },
+});
```
Comment (itisAliRH) [inline client/src/components/History/HistoryScrollList.vue]: It is more straightforward to use the generic types:
https://vuejs.org/guide/typescript/composition-api#typing-component-props

**Comment by ahmedhamidawan** [comment]
Failing Toolshed tests seem unrelated...

**Review by itisAliRH — APPROVED** [review]
Looks good to me, thanks! @ahmedhamidawan

File: client/src/components/History/CurrentHistory/HistoryPanel.vue [hunk 4]
```diff
@@ -41,31 +42,23 @@ interface BackendFilterError {
...
-        type: Boolean,
-        default: true,
-    },
-    filterable: {
-        type: Boolean,
-        default: false,
-    },
+interface Props {
+    listOffset?: number;
+    history: Record<string, any> & GetComponentPropTypes<typeof HistoryCounter>["history"];
```
Comment (davelopez) [inline client/src/components/History/CurrentHistory/HistoryPanel.vue]: Minor thing, but we can import `import { HistorySummary } from "@/api";` and replace the type here to make it more readable.

```suggestion
    history: HistorySummary;
```

**Review by davelopez — APPROVED** [review]
Thank you! Cool stuff!

File: client/src/components/History/CurrentHistory/HistoryPanel.vue (hunk 4 above)
Comment (ahmedhamidawan) [reply client/src/components/History/CurrentHistory/HistoryPanel.vue]: Done, thank you!

**Pull request merged**